pandas/core/generic.py

```diff
@@ -3749,8 +3752,13 @@ class   animal   locomotion
             )
 
         else:
-            result = self.iloc[loc]
-            result.index = new_index
+            if axis == 0:
+                result = self.iloc[loc]
+                result.index = new_index
+            else:
+                result = (
+                    self.iloc[:, loc] if not is_scalar(loc) else self.iloc[:, [loc]]
```

Comment:
worth making is_scalar another case in the eliif (I think it will read simpler that way)

Reply:
Yep, reads better, thx.